server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -160,9 +163,11 @@ final class TransportHandshaker {
      */
 
     private static final Logger logger = LogManager.getLogger(TransportHandshaker.class);
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(logger.getName());
 
     static final TransportVersion V7_HANDSHAKE_VERSION = TransportVersion.fromId(6_08_00_99);
     static final TransportVersion V8_HANDSHAKE_VERSION = TransportVersion.fromId(7_17_00_99);
+    static final TransportVersion V8_18_HANDSHAKE_VERSION = TransportVersions.ML_INFERENCE_IBM_WATSONX_RERANK_ADDED;
```

Comment:
This isn't a _handshake_ version. Handshake versions are special, there's only 3 of them in scope here. It's just a regular `TransportVersion`. I'd rather it wasn't in this list.